CHANGELOG.next.asciidoc

```diff
@@ -44,6 +44,10 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Update Salesforce module to use new Salesforce input. {pull}37509[37509]
 - Tag events that come from a filestream in "take over" mode. {pull}39828[39828]
 - Fix high IO and handling of a corrupted registry log file. {pull}35893[35893]
+- Journald: removed configuration options `include_matches.or`, `include_matches.and`, `backoff`, `max_backoff`, `cursor_seek_fallback`. {pull}40061[40061]
+- Journald: `include_matches.match` now behaves in the same way as matchers in `journalctl`. {pull}40061[40061]
```

Comment:
If you can link to the new matcher behavior and/or highlight some differences that should help. What should users do as a consequence of this change?

Reply:
You mean link to the latest version of the documentation using the anchors introduced by this PR?

The TL;DR: is users should update their configurations. 

Long version: there is a small chance they managed to create a filter that is not currently supported, in that case the best option is to user processors to filter out the events or wait for another release where we can extend the behaviour of the matchers to match 100% what `journalctl` exposes.

Reply:
We will need to very clearly document this, and have a follow up issue to get to parity with what `journalctl` exposes.

Reply:
Issue created: https://github.com/elastic/beats/issues/40185

Reply:
Given the somehow odd syntax/support provided by the journalctl matchers I like the idea of just exposing a string or array that we pass directly to `journalctl` and document it. This way we're not re-inventing the wheel just to parse our config and pass it as is to `journalctl`.

See some somehow odd logic expressions we can create with the current implementation:
https://github.com/elastic/beats/blob/45c2759d0ea1145439d399a699c9f35ae0c1eb47/filebeat/input/journald/input_test.go#L255-L286

Even the `journalctl` documentation is not clear on how an arbitrary number of matchers with the `+` operator (or multiple) are executed:
> Finally, the character "+" may appear as a separate word between
       other terms on the command line. This causes all matches before and after to be combined in a disjunction (i.e. logical OR).


Anyways, we can discuss this in more depth in the issue I linked above.